### PR TITLE
Make Guard take a builder

### PR DESCRIPTION
**Pull request by tomgilder**

### CODE REVIEW

File: example/book_store/lib/main.dart [hunk 1]
```diff
@@ -53,14 +53,14 @@ final routeMap = RouteMap(
           validate: (info, context) => booksDatabase.books.any(
             (book) => book.id == info.pathParameters['id'],
           ),
-          child: MaterialPage(
+          pageBuilder: () => MaterialPage(
```
Comment (chimon2000) [inline example/book_store/lib/main.dart:56]: Should it just be `builder` since there is only one builder?

Comment (tomgilder) [reply example/book_store/lib/main.dart:56]: Good question - my thinking with `pageBuilder` was to make it clear it needs to return a page and not a widget. But I guess it wouldn't compile anyway, and it should be pretty obvious from the error

Comment (ryan-sf) [reply example/book_store/lib/main.dart:56]: Yes, I was thinking that the code wouldn't run since the developer would get an error.

Comment (tomgilder) [reply example/book_store/lib/main.dart]: Done

**Pull request merged**